### ISSUE TITLE
Unit support in DataTag

### ISSUE THREAD

**Issue by jgrewe** [G-Node/nix#227]
The DataTag is used to annotate multiple points in an n-dimensional dataset. These positions are stored in a 2-dimensional DataArray. The first dimension of this represents the number of positions, each entry along the second one is a pointer in the respective dimension of the data.  So far, these pointers are interpreted in the same units as the data. In SimpleTag, however, we support to point into data with arbitrary, yet scalable units. For example, a dimension of the data may represent time and is given in seconds. In a SimpleTag one could refer to a certain point along this dimension also in milliseconds. From the users point of view this is desired behavior since I simply need to look at the tag to understand it. While one would need to dig into the data definitions in the case of the DataTag.

Long story short: For consistency we decided to include unit support also in DataTags. That means that there is a list of strings in DataTag that takes this information. Problems: We need to support nan, or similar constructs, to indicate that there is no unit, e.g. for references along set dimensions. Secondly, the validator must check  the validity of the references and compatibility of the provided units.
